# Reads mapping on the reference genome
```sh
# Convert GFF into GTF
gffread red.gene.gff -T -o red.gene.gtf

# Build index of STAR
STAR --runMode genomeGenerate --genomeDir starRed_genome --runThreadN 16 --genomeFastaFiles red.contig.fa --sjdbGTFfile red.gene.gtf --sjdbOverhang 149 --sjdbGTFfeatureExon CDS --genomeSAindexNbases 13

# Map data of each sample (Glv1 is the example sample name)
STAR --quantMode GeneCounts --genomeDir starRed_genome --runThreadN 16 --outFilterMismatchNmax 2 --readFilesIn Glv1_R1.clean.fq.gz Glv1_R2.clean.fq.gz --readFilesCommand zcat --outFileNamePrefix Glv1.star --outFilterMultimapNmax 1 --outSAMtype BAM SortedByCoordinate --outReadsUnmapped Glv1.unmapped

# Summarize reads count
ls *.starReadsPerGene.out.tab > z.all_counts.lst
perl sum_count.pl z.all_counts.lst > z.all_counts.lst.txt

# Differentially expressed genes (DEGs) finding (based on DESeq2)
R CMD BATCH DEG_Red_vs_Both.R DEG_Red_vs_Both.Rout

# Create a table for significantly DEGs
less DEG_Red_vs_Both.csv | awk -F',' '(($3 > 1) || ($3 < -1)) && ($7 < 0.05)' > DEG_Red_vs_Both.csv.sig

# Compute transcript per million (TPM)
perl tpm_calculator.pl red.gene.len z.all_counts.lst.txt > z.all_counts.lst.txt.TPM

# Draw heatmap for gene expression
R CMD BATCH ADH_pheatmap.R ADH_pheatmap.Rout

# Combine reads count of A. tricolor cv. Green
perl combine_green_count.pl z.all_counts.lst.txt z.green_counts.lst.txt > z.red_green.counts.txt
```

# Gene ontology (GO) enrichment analysis
```sh
# Download GO database
wget http://purl.obolibrary.org/obo/go/go-basic.obo

# Convert format for the database
python3 get_go_term.py go-basic.obo

# Get GOs for gene set
interproscan-5.52-86.0/interproscan.sh -i ./red.pep.fa -dp -goterms -iprlookup -pa
iprscan_parser.pl red.pep.fa.tsv > red.pep.parser.out 2> red.pep.parser.err
perl gene_go.pl red.pep.fa.tsv.gene.wego > red.pep.fa.tsv.gene.wego.go

# Get DEGs (expressed more or less)
less -SN Rgreen_Rboth_DEGs.csv.sig | awk -F',' '$3 < 0' > Rgreen_Rboth_DEGs.csv.sig.higer_both

# Do GO enrichment by clusterProfilier
R CMD BATCH go_enrichment.R go_enrichment.Rout
```
